### PR TITLE
[TT-13109]Generate New Swagger and Update Validator for Gateway

### PR DESCRIPTION
**Pull request by Keithwachira**
## **User description**

This pull request contains :
1. The new gateway swagger
2.  A new linter
3.  And a readme of how to generate the swagger 
.

.This provides the documentation on how you can generate the gateway Open Api specification. It contains a read me with all the steps to generate the OAS.

The OAS code is in this pr : https://github.com/TykTechnologies/tyk/pull/6123
[TT-13109]

___

## **Type**
Documentation

___

## **Description**
- Added a comprehensive README.md in the `tyk-api-documentation` directory.
- The documentation includes details on the version of OAS used, the tooling (Redocly), and the library (`openapi-go`) for OAS generation.
- Step-by-step instructions are provided for generating the Swagger.yml file using a make command in the specified repository branch.
- Describes the file structure within the Swagger directory, explaining the organization of files by OAS tags and the roles of various files in the generation process.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Add Documentation for Generating Tyk API Swagger File</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-api-documentation/README.md
<li>Introduced a new README.md for Tyk API Documentation.<br> <li> Detailed the process and tools used to generate the Open API <br>Specification (OAS).<br> <li> Provided step-by-step instructions on how to generate the Swagger.yml <br>file.<br> <li> Explained the file structure within the Swagger directory.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/6231/files#diff-25581cb9574ab0d44d58541ab1ead676194ddcb9dfe7ba4687197f9673df6093">+33/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

[DX-1272]: https://tyktech.atlassian.net/browse/DX-1272?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TT-13109]: https://tyktech.atlassian.net/browse/TT-13109?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/redocly-swagger-lint.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    branches:
+      - master
+      - release-**
+jobs:
+  redocly_validator:
+    runs-on: ubuntu-latest
+    name: Validate the swagger with redocly cli
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (letzya) [inline .github/workflows/redocly-swagger-lint.yml:12]: can we use the v4?

File: tyk-api-documentation/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+The version of Open Api specification used is currently 3.0.3
+
+Currently, we use redocly for linting the OAS. You can install Redocly from (https://github.com/Redocly/redocly-cli)
+
+
+## How the OAS was generated.
+
+We used the golang library [openapi-go](https://github.com/swaggest/openapi-go) because:
+1. It supports producing  OASn for version 3. (Golang doesn't to many libraries to produce OAS three)
```
Comment (letzya) [inline tyk-api-documentation/README.md:16]:
```suggestion
1. It supports producing  OAS for version 3. (Golang doesn't have too many libraries that produce OAS 3)
```

Comment (letzya) [reply tyk-api-documentation/README.md:16]: @Keithwachira which v3 is it? 3.1?

**Review by lghiur — APPROVED** [review]

File: .redocly.lint-ignore.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# This file instructs Redocly's linter to ignore the rules contained for specific parts of your API.
+# See https://redoc.ly/docs/cli/ for more information.
+swagger.yml:
```
Comment (titpetric) [inline .redocly.lint-ignore.yaml:3]: Can this include more comments/docs as to why this is configured as is? Why do we need ignores?

File: .github/workflows/lint-swagger.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@ jobs:
     name: Swagger Editor Validator Remote
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
```
Comment (titpetric) [inline .github/workflows/lint-swagger.yml:13]: This still runs the old linter, if you can, sync the structure with analytics:
- lint-swagger.yml only (no redocly...)

File: tyk-gateway-oas-generation-documentation/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Tyk API Documentation
```
Comment (titpetric) [inline tyk-gateway-oas-generation-documentation/README.md:1]: Hmmm, this needs a /docs addition. We currently don't have that folder, but, suggesting `/docs/swagger.md` at this point (we may change this in future as conventions get adjusted/considered).

Docs has some issues:

- consistency (semver is `vN.N.N`, several `v` prefixes omitted)...
- language, `add it to it specific tag` - maybe just re-read it or throw it through a grammar check :)
- spacing (L5/headings should follow with an empty line like L1, 13, 20, 31)

**Review by titpetric — APPROVED** [review]
:clap:

**Pull request merged**